corehq/apps/cloudcare/static/cloudcare/js/utils.js

```diff
@@ -148,47 +148,59 @@ hqDefine('cloudcare/js/utils', [
             "cloudcare/js/formplayer/app",
-            "hqwebapp/js/toggles",
             "cloudcare/js/formplayer/layout/views/progress_bar",
-        ], function (FormplayerFrontend, toggles, ProgressBar) {
-            if (toggles.toggleEnabled('USE_PROMINENT_PROGRESS_BAR')) {
-                const progressView = ProgressBar({
-                    progressMessage: gettext("Loading..."),
-                });
-                if (!FormplayerFrontend.regions) {
-                    FormplayerFrontend.regions = getRegionContainer();
-                }
-                $('#breadcrumb-region').css('z-index', '0');
-                const loadingElement = FormplayerFrontend.regions.getRegion('loadingProgress');
-                loadingElement.show(progressView);
-                let currentProgress = 10;
-                progressView.progressEl.find('.progress').css("height", "12px");
-                progressView.progressEl.find('.progress-container').css("width", "50%");
-                progressView.progressEl.find('.progress-title h1').css("font-size", "25px");
-                progressView.progressEl.find('#formplayer-progress ').css("background-color", "rgba(255, 255, 255, 0.7)");
-                progressView.setProgress(currentProgress, 100, 200);
-                sessionStorage.progressIncrementInterval = setInterval(function () {
-                    if (currentProgress <= 100) {
-                        progressView.setProgress(currentProgress, 100, 200);
-                        currentProgress += 1;
+        ], function (FormplayerFrontend, ProgressBar) {
+            setTimeout(function () {
+                const formplayerQueryInProgress = sessionStorage.formplayerQueryInProgress && JSON.parse(sessionStorage.formplayerQueryInProgress);
+                if (formplayerQueryInProgress) {
+                    const progressView = ProgressBar({
+                        progressMessage: gettext("Loading..."),
+                    });
+                    if (!FormplayerFrontend.regions) {
+                        FormplayerFrontend.regions = getRegionContainer();
                     }
-                }, 250);
+                    $('#breadcrumb-region').css('z-index', '0');
+                    const loadingElement = FormplayerFrontend.regions.getRegion('loadingProgress');
+                    loadingElement.show(progressView);
+                    let currentProgress = 10;
+                    progressView.progressEl.find('.progress').css("height", "12px");
+                    progressView.progressEl.find('.progress-container').css("width", "50%");
+                    progressView.progressEl.find('.progress-title h1').css("font-size", "25px");
+                    progressView.progressEl.find('#formplayer-progress ').css("background-color", "rgba(255, 255, 255, 0.7)");
+                    progressView.setProgress(currentProgress, 100, 200);
+                    sessionStorage.progressIncrementInterval = setInterval(function () {
```

Comment:
could `progressIncrementInterval` also be moved into `progress_bar.js`? I think you would also be able to access this interval in `hideLoading` for clearing but i'm not entirely sure.

Reply:
You would still have access for clearing, but I would worry about muddying up the functionality in `progress_bar.js` since it's being used in other places like https://github.com/dimagi/commcare-hq/blob/72596d653db925d49bd7e878cbe500c2d4c9dee9/corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js#L568